pkg/manager/internal.go

```diff
@@ -485,8 +485,8 @@ func (cm *controllerManager) engageStopProcedure(stopComplete <-chan struct{}) e
 				cm.internalCancel()
 			})
 			select {
-			case err, ok := <-cm.errChan:
-				if ok {
+			case err := <-cm.errChan:
+				if !errors.Is(err, context.Canceled) {
```

Comment:
Why not keep checking the `ok` value here?

Reply:
I don't see any calls to close `errChan`, so `ok` is always true (today).

Also, in my experience, reading from a closed channel in `select` is for some kind of control flow (break or return or so), or the channel is assigned `nil` so that it is not selected again.

Reply:
Sounds okay to me. I guess if we ever start closing the channel we have to check all usages of the channel anyway.